File: Add Code Here/C++ Leetcode Solutions/189-rotate-array.cpp
class Solution {
public:
    void rotate(vector<int>& nums, int k) {
        
         k=k%nums.size();
        int tmp=nums.size()-k;
        
        reverse(nums.begin(),nums.begin()+tmp);
        reverse(nums.begin()+tmp,nums.end());
        
        reverse(nums.begin(),nums.end());
    }
};

File: Add Code Here/LeetCode Problems/189. Rotate Array.java
class Solution {
    public void rotate(int[] nums, int k) {
        k %= nums.length;
        int n = nums.length;
        reverseNum(nums,0,n-1);
        reverseNum(nums,0,k-1);
        reverseNum(nums,k,n-1);
    }
    public void reverseNum(int[] nums, int start, int end) {
        while(start < end) {
            int temp = nums[start];
            nums[start] = nums[end];
            nums[end] = temp;
            start++;
            end--;
        }
    }
}
